ovssubnet/api/types.go

```diff
@@ -65,3 +69,23 @@ type NamespaceEvent struct {
 	Type EventType
 	Name string
 }
+
+type ServiceProtocol string
+
+const (
+	TCP ServiceProtocol = "TCP"
+	UDP ServiceProtocol = "UDP"
+)
+
+type Service struct {
+	Name      string
+	Namespace string
+	IP        string
+	Protocol  ServiceProtocol
```

Comment:
Yes, AddServiceOFRules() adds either ",tcp" or ",udp" to the flow rule based on that